lib/brain.js

```diff
@@ -2045,7 +2045,8 @@ Brain.prototype.enableBillValidator = function enableBillValidator () {
 
 Brain.prototype.disableBillValidator = function disableBillValidator () {
   emit('billValidatorOff')
-  this.billValidator.disable()
+  if (this.billValidator.initialized)
```

Comment:
Since this is a low level error, I think it would be better to handle it in the billValidator module. It would be better if we didn't have to ``check billValidator.initialized`` from outside the billValidator module.

One reason for this is that we can crash the app by calling ``this.billValidator.disable()`` anywhere in the code if we're not careful to check that it's initialized. If we put the logic in billValidator, we should never be able to crash the app by calling it.

Reply:
Hmm, if we're to solve this from the inside how should we deal with functions like [`isCashOutBox()`](https://github.com/lamassu/lamassu-machine/blob/dev/lib/id003/id003.js#L51) which either return `true` or `false` to the calling code? The "proper" way would be to throw an exception because the code that evaluates that hasn't been initialized, but in that case we also have to update the client code to deal with such cases.

Reply:
Yeah, an exception makes sense. An exception will be thrown anyway by nodejs, we'd just be making it more clear what the error is by catching and re-throwing. The important part is that higher level functions shouldn't have to worry about internal state of the functions they're calling.